cmake/sdlchecks.cmake

```diff
@@ -82,29 +82,33 @@ endmacro()
 # - SDL_ALSA_SHARED opt
 # - HAVE_SDL_LOADSO opt
 macro(CheckALSA)
-  if(SDL_ALSA)
-    CHECK_INCLUDE_FILE(alsa/asoundlib.h HAVE_ASOUNDLIB_H)
-    if(HAVE_ASOUNDLIB_H)
-      CHECK_LIBRARY_EXISTS(asound snd_pcm_recover "" HAVE_LIBASOUND)
-    endif()
-    if(HAVE_LIBASOUND)
-      set(HAVE_ALSA TRUE)
-      file(GLOB ALSA_SOURCES ${SDL2_SOURCE_DIR}/src/audio/alsa/*.c)
-      list(APPEND SOURCE_FILES ${ALSA_SOURCES})
-      set(SDL_AUDIO_DRIVER_ALSA 1)
-      if(SDL_ALSA_SHARED AND NOT HAVE_SDL_LOADSO)
-        message_warn("You must have SDL_LoadObject() support for dynamic ALSA loading")
-      endif()
-      FindLibraryAndSONAME("asound")
-      if(SDL_ALSA_SHARED AND ASOUND_LIB AND HAVE_SDL_LOADSO)
-        set(SDL_AUDIO_DRIVER_ALSA_DYNAMIC "\"${ASOUND_LIB_SONAME}\"")
-        set(HAVE_ALSA_SHARED TRUE)
-      else()
-        list(APPEND EXTRA_LIBS asound)
-      endif()
-      set(HAVE_SDL_AUDIO TRUE)
+    if(SDL_ALSA)
+        find_package(ALSA MODULE)
```

Comment:
Using imported targets is useful to keep the installed SDL prefix portable (=containing no paths of the build systems).
I took a shot in https://github.com/libsdl-org/SDL/pull/6084, but providing a module for each dependency is overkill.

WDYT about the following approach?
```cmake
find_package(ALSA)
if(ALSA_FOUND)
    sdl_create_alsa_imported_target)()
    # make sure SDL2/SDL2-static link to ALSA::ALSA such that it ends up in `SDL2staticTargetscmake
    # in SDL2Config.cmake, we need to make sure to do `find_package(ALSA)` and `sdl_create_alsa_imported_target()`
endif()
...
function(sdl_create_alsa_imported_target)
    if(NOT TARGET ALSA::ALSA)
        add_Library(ALSA::ALSA INTERFACE IMPORTED)
        set_target_properties(ALSA::ALSA
            PROPERTIES
                INTERFACE_INCLUDE_DIRECTORIES "${ALSA_INCLUDE_DIR}"
                INTERFACE_LINK_DIRECTORIES "${ALSA_LIBRARY_DIRS}"
                INTERFACE_LINK_DIRECTORIES "${ALSA_LIBRARIES}"
        )
    endif()
endfunction()
```

If we add `sdl_create_alsa_imported_target` to a cmake file that is also installed, the cmake package is completely portable.

WDYT?

Reply:
I like that approach. Although I combined the find_package and sdl_create_alsa_imported_target into sdlFindALSA.

Reply:
I also like your solution. I'm going to port it to SDL3 once this pr finishes.